test/unit/model/consent.test.ts

```diff
@@ -0,0 +1,254 @@
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs',
+  credentialPayload: 'dwuduwd&e2idjoj0w'
+}
+
+/*
+ * Consent Resource Model Unit Tests
+ */
+describe('src/model/consent', (): void => {
+  let Db: Knex
+  let consentDB: ConsentDB
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+    await Db.raw('PRAGMA foreign_keys = ON')
+
+    consentDB = new ConsentDB(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  describe('register', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+    })
+
+    it('adds consent with partial info to the database', async (): Promise<void> => {
+      // Action
+      const numInserted: number = await consentDB.register(partialConsent)
+
+      expect(numInserted).toEqual(1)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: partialConsent.id
+        })
+
+      expect(consents[0]).toEqual({
+        id: '1234',
+        initiatorId: 'pisp-2342-2233',
+        participantId: 'dfsp-3333-2123',
+        createdAt: expect.anything(),
+        credentialId: null,
+        credentialType: null,
+        credentialStatus: null,
+        credentialPayload: null,
+        credentialChallenge: null
+      })
+    })
+
+    it('returns an error on adding a consent with existing consentId', async (): Promise<void> => {
+      // Action
+      await consentDB.register(partialConsent)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: partialConsent.id
+        })
+
+      // Consent has been added
+      expect(consents[0]).toEqual({
+        id: '1234',
+        initiatorId: 'pisp-2342-2233',
+        participantId: 'dfsp-3333-2123',
+        createdAt: expect.anything(),
+        credentialId: null,
+        credentialType: null,
+        credentialStatus: null,
+        credentialPayload: null,
+        credentialChallenge: null
+      })
+
+      // Fail primary key constraint
+      await expect(consentDB.register(partialConsent))
+        .rejects.toThrowError()
+    })
+  })
+
+  describe('updateCredentials', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+    })
+
+    it('updates credentials for existing consent', async (): Promise<void> => {
+      // Inserting record to update
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      // Action
+      await consentDB.updateCredentials(completeConsent)
+
+      // Assertion
+      const consents: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: completeConsent.id
+        })
+
+      expect(consents[0].createdAt).toEqual(expect.anything())
```

Comment:
Can we check that this is an instance of `Date` ?

Reply:
Design decision required about MySQL timestamp return and implementation of Data methods.